packages/virtual/src/index.tsx

```diff
@@ -1,71 +1,123 @@
 import { For, createSignal } from "solid-js";
 import type { Accessor, JSX } from "solid-js";
+import type { DOMElement } from "solid-js/jsx-runtime";
+import { access } from "@solid-primitives/utils";
+import type { MaybeAccessor } from "@solid-primitives/utils";
+
+type VirtualListConfig<T extends readonly any[]> = {
+  items: MaybeAccessor<T | undefined | null | false>;
+  rootHeight: MaybeAccessor<number>;
+  rowHeight: MaybeAccessor<number>;
+  overscanCount?: MaybeAccessor<number>;
+};
+
+type VirtualListReturn<T extends readonly any[]> = [
+  {
+    containerHeight: Accessor<number>;
+    viewerTop: Accessor<number>;
+    visibleItems: Accessor<T>;
+  },
+  onScroll: (
+    e: Event & {
+      target: DOMElement;
+    },
+  ) => void,
```

Comment:
```suggestion
  onScroll: (e: Event) => void,
```
I don't see any point in containing the callback type more. I'm not even sure what it that supposed to do.

Reply:
This surprised me too, but it's actually necessary; otherwise you get a typescript error that `e.target` is possibly null on line 62. I don't think it constrains the type in any harmful way, you should really only be attaching onScroll as an event handler for something that emits events with a `DOMElement` `target`.

Reply:
Then you cannot use it with `addEventListener`
While we can just add a null check